appointment-svc/service/db-service/AppointmentRepository.ts

```diff
@@ -109,6 +109,34 @@ class AppointmentRepository {
 
     return isPendingStatus;
   }
+
+  async getAvailableAppointment(currentDate: Date) {
+    const targetDate = new Date(currentDate);
+    targetDate.setDate(currentDate.getDate() + 1);
+
+    const availableAppointment = await prisma.appointment.findFirst({
+      where: {
+        open_to_earlier: true,
+        isPending: false,
+        date: {
+          gte: currentDate,
+          lte: targetDate,
+        },
+      },
+    });
+
+    if (availableAppointment) {
+      await this.changeIsPendingValue(
+        availableAppointment.id,
+        availableAppointment.isPending
+      );
+      console.log(availableAppointment);
+    } else {
+      console.log('No available appointment at this time');
+    }
```

Comment:
Also, I think you should extract this logic and put it in the compiler so you don't call two functions as one, it will make it easier to track errors and test.
Also, as a second argument for changePendingValue, you should pass true or false depending on what value you want to assign.
Right now, you get the isPending from the avaliableAppointment, which is false, and just pass false as a prop so after the call, it will stay false.

Reply:
ah i understand. fixed it

Reply:
I'm sorry, made a typo, i meant 'controller'